lib/spack/spack/binary_distribution.py

```diff
@@ -1215,6 +1214,9 @@ def _build_tarball(
     """
     Build a tarball from given spec and put it into the directory structure
     used at the mirror (following <tarball_directory_name>).
+
+    This method raises NoOverwriteException when force=False and the tarball or
+    spec.json file already exist in the buildcache.
```

Comment:
Not a fault of this PR, but as we are at it, can we add a description of each argument in the docstring (and type-hints)?